File: morning_exercises/day12.cpp
#include <vector>
#include <iostream>

int binarySearch(std::vector<int> input, int left, int right, int val) {
    if (left>right) return -1;
    int mid = (left+right)/2;
    if (val == input[mid]) return mid;
    else if (val<input[mid]) return binarySearch(input, left, mid-1, val);
    return binarySearch(input, mid+1, right, val);
}

int main() {
    std::vector<int>input = {1,2,3,4,5,6,7,8,9,10};
    std::cout << binarySearch(input, 0, 9, 9);
}

File: morning_exercises/day12.ts
function binarySearch(input: number[], left: number, right: number, val: number): number {
    if (left>right) return -1;
    let mid: number = Math.floor((left+right)/2);
    if (val===input[mid]) return mid;
    else if (val<input[mid]) return binarySearch(input, left, mid-1, val);
    return binarySearch(input, mid+1, right, val);
}

{
    let input = [2,4,7,9,13,17];
    console.log(binarySearch(input, 0, input.length-1, 7));
}